src/extras/PMREMGenerator.js

```diff
@@ -205,8 +222,18 @@ class PMREMGenerator {
 
 	_fromTexture( texture, renderTarget ) {
 
+		if ( texture.mapping === CubeReflectionMapping || texture.mapping === CubeRefractionMapping ) {
+
+			this._setSize( texture.image.length === 0 ? 16 : texture.image[ 0 ].width ?? texture.image[ 0 ].image.width );
```

Comment:
@Mugen87 This is hideous and hacky, and I've love for there to be a better way. I really wish textures had their own `width` and `height` members, because as it stands, finding that info is totally different for textures, cubemaps, data textures, textures created by renderTargets, etc. I'm not sure what the best path forward for three.js is, but this was the best I could manage for now to get this working. Also, I'm falling back to 16 because that's what the WebXR cubemap is. I couldn't find its size anywhere at all; am I missing something?

Reply:
> I couldn't find its size anywhere at all; am I missing something?

I did not find a note about the size in the documentation.

> I'm not sure what the best path forward for three.js is, but this was the best I could manage for now to get this working. 

Yeah, it's inconvenient right now to cover all possible texture permutations. The main problem is that cube textures can hold an array of images _or_ an array of (data) textures. I'm not sure yet but one solution could be to wrap the images of a cube texture in instances of `Texture` to unify the access.

Reply:
Maybe, but I was thinking just add `width` and `height` members to the base texture class  and  fill them in appropriately so that we  don't have to dig into the `.image` at all. 

And my point with the WebXR environment cube map is that the three.js code knows the size (it gets an array back from the API), but I don't think it saves that size anywhere that  we  as users of three.js can  retrieve. That feels like a  bug, but a bug  in the API shape that a texture `width` and `height` could solve.